### PR TITLE
[SPARK-12234] [SparkR] Fix ```subset``` function error when only set ```select``` argument

### PR DESCRIPTION
**Pull request by yanboliang**
Fix `subset` function error when only set `select` argument. Please refer to the [JIRA](https://issues.apache.org/jira/browse/SPARK-12234) about the error and how to reproduce it.

cc @sun-rui @felixcheung @shivaram

### CODE REVIEW

**Comment by felixcheung** [comment]
cool, looks good

File: R/pkg/R/DataFrame.R [hunk 1]
```diff
@@ -1209,7 +1209,11 @@ setMethod("[", signature(x = "DataFrame", i = "Column"),
 #' }
 setMethod("subset", signature(x = "DataFrame"),
           function(x, subset, select, ...) {
-            x[subset, select, ...]
+            if (missing(subset)) {
```
Comment (felixcheung) [inline R/pkg/R/DataFrame.R]: best if we could add an example above

File: R/pkg/R/DataFrame.R [hunk 2]
```diff
@@ -1206,10 +1206,15 @@ setMethod("[", signature(x = "DataFrame", i = "Column"),
 #'   df[df$age %in% c(19, 30), 1:2]
 #'   subset(df, df$age %in% c(19, 30), 1:2)
 #'   subset(df, df$age %in% c(19), select = c(1,2))
+#'   subset(df, select = c(1,2))
 #' }
 setMethod("subset", signature(x = "DataFrame"),
           function(x, subset, select, ...) {
-            x[subset, select, ...]
+            if (missing(subset)) {
+                x[, select, ...]
```
Comment (sun-rui) [inline R/pkg/R/DataFrame.R]: 2-space ident

**Comment by sun-rui** [comment]
also could you state in the method description that subset parameter is optional?

**Comment by sun-rui** [comment]
LGTM

**Comment by shivaram** [comment]
LGTM. Thanks @yanboliang - Merging this

**Pull request closed without merging**